Dockerfile

```diff
@@ -0,0 +1,203 @@
+WORKDIR /tensorflow
+RUN git checkout r1.6
+
+
+# GPU Environment Setup
+ENV TF_NEED_CUDA 1
+ENV CUDA_TOOLKIT_PATH /usr/local/cuda
+ENV CUDA_PKG_VERSION 9-0=9.0.176-1
+ENV CUDA_VERSION 9.0.176
+ENV TF_CUDA_VERSION 9.0
+ENV TF_CUDNN_VERSION 7.1.1
+ENV CUDNN_INSTALL_PATH /usr/lib/x86_64-linux-gnu/
+ENV TF_CUDA_COMPUTE_CAPABILITIES 6.0
+
+# Common Environment Setup
+ENV TF_BUILD_CONTAINER_TYPE GPU
+ENV TF_BUILD_OPTIONS OPT
+ENV TF_BUILD_DISABLE_GCP 1
+ENV TF_BUILD_ENABLE_XLA 0
+ENV TF_BUILD_PYTHON_VERSION PYTHON2
+ENV TF_BUILD_IS_OPT OPT
+ENV TF_BUILD_IS_PIP PIP
+
+# Other Parameters
+ENV CC_OPT_FLAGS -mavx -mavx2 -msse4.1 -msse4.2 -mfma
+ENV TF_NEED_GCP 0
+ENV TF_NEED_HDFS 0
+ENV TF_NEED_JEMALLOC 1
+ENV TF_NEED_OPENCL 0
+ENV TF_CUDA_CLANG 0
+ENV TF_NEED_MKL 0
+ENV TF_ENABLE_XLA 0
+ENV PYTHON_BIN_PATH /usr/bin/python2.7
+ENV PYTHON_LIB_PATH /usr/lib/python2.7/dist-packages
+
+# << END Configure Tensorflow Build
+
+
+
+
+# >> START Configure Bazel
+
+# Running bazel inside a `docker build` command causes trouble, cf:
+#   https://github.com/bazelbuild/bazel/issues/134
+# The easiest solution is to set up a bazelrc file forcing --batch.
+RUN echo "startup --batch" >>/etc/bazel.bazelrc
+# Similarly, we need to workaround sandboxing issues:
+#   https://github.com/bazelbuild/bazel/issues/418
+RUN echo "build --spawn_strategy=standalone --genrule_strategy=standalone" \
+    >>/etc/bazel.bazelrc
+
+# Put cuda libraries to where they are expected to be
+RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1
+RUN cp /usr/include/cudnn.h /usr/local/cuda/include/cudnn.h
+
+# Set library paths
+ENV LD_LIBRARY_PATH $LD_LIBRARY_PATH:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/lib/x86_64-linux-gnu/:/usr/local/cuda/lib64/stubs/
+
+# << END Configure Bazel
+
+
+
+
+# Copy DeepSpeech repo contents to container's /DeepSpeech
+COPY . /DeepSpeech/
+
+WORKDIR /DeepSpeech
+
+RUN pip --no-cache-dir install -r requirements.txt
+
+# Link DeepSpeech native_client libs to tf folder
+RUN ln -s /DeepSpeech/native_client /tensorflow
+
+
+
+
+# >> START Build and bind
+
+WORKDIR /tensorflow
+
+# passing LD_LIBRARY_PATH is required cause Bazel doesn't pickup it from environment
+
+# Build LM Prefix Decoder
+RUN bazel build -c opt --copt=-O3 --config=cuda --copt="-D_GLIBCXX_USE_CXX11_ABI=0" //native_client:libctc_decoder_with_kenlm.so  --verbose_failures --action_env=LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+
+# Build DeepSpeech
+RUN bazel build --config=monolithic -c opt --copt=-O3 --copt="-D_GLIBCXX_USE_CXX11_ABI=0" --copt=-fvisibility=hidden //native_client:libdeepspeech.so //native_client:deepspeech_utils //native_client:generate_trie --verbose_failures --action_env=LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+
+
+# Build TF pip package
+RUN bazel build --config=opt --config=cuda  --copt=-msse4.2 //tensorflow/tools/pip_package:build_pip_package --verbose_failures --action_env=LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
```

Comment:
Building the Python package takes a lot of time, there's likely no value in doing it :)

Reply:
To experiment with different options of TF like XLA or FP16

Reply:
@GeorgeFedoseev That comment holds, there should be no value into building the pip wheel from scratch rather than using the one from tensorflow, except wasting CPU and memory

Reply:
That makes sense.